src/components/SchemaViewer.svelte

```diff
@@ -11,24 +11,38 @@
       .split(" ")
       .filter((t) => t.length > 0);
 
-    const addVisibility = (node) => {
+    const addVisibility = (node, parentNodeNames = [`${node.name}`]) => {
       let modifiedNode = node;
+      let allParentNames = [...parentNodeNames];
+      let { name, parentNames, fields } = modifiedNode;
 
-      if (modifiedNode.fields) {
-        modifiedNode.fields.forEach(addVisibility);
+      if (fields) {
+        allParentNames = parentNames ? `${parentNames}.${name}` : name;
+
+        fields.forEach((field) => {
+          let modifiedNodeField = field;
+          const parentNameArr = [];
+
+          modifiedNodeField.parentNames = allParentNames;
+          parentNameArr.push(`${allParentNames}.${modifiedNodeField.name}`);
+
+          return addVisibility(modifiedNodeField, parentNameArr);
+        });
       }
       modifiedNode.visible =
         filterTerms.length === 0 ||
-        filterTerms.every((term) => modifiedNode.name.includes(term));
-      modifiedNode.childrenVisible = modifiedNode.fields
-        ? modifiedNode.fields.some(
-            (child) => child.visible || child.childrenVisible
-          )
+        filterTerms.every((term) =>
+          parentNames
+            ? parentNames.includes(term) || name.includes(term)
+            : name.includes(term)
+        );
+      modifiedNode.childrenVisible = fields
+        ? fields.some((child) => child.visible || child.childrenVisible)
         : undefined;
       return modifiedNode;
     };
 
-    nodesWithVisibility = nodes.map(addVisibility, filterTerms);
+    nodesWithVisibility = nodes.map((node) => addVisibility(node), filterTerms);
```

Comment:
I did this because I get this error `number 0 is not iterable` .When addVisibility is first called, the index of the node is passed into the `parentNodeNames` which cannot be iterated